### ISSUE TITLE
Not working as of Jodel 5.32.1

### ISSUE THREAD

**Issue by 7passd0** [Unbrick/r2HmacExtractor#2]
Hello, it looks like the HMAC secret returned is no longer working. I double checked and the secret calculated by this script is the same as the one calculated with OJOC-keyhack, so either Jodel changed something in the way secret is calculated (and the current method of doing so returns a wrong result), or they changed how the HMAC payload is build.

**Comment by Telios32** [comment]
+1 
I can confirm that your suspicions are correct. Jodel definitely changed something about her APP.

**Comment by Unbrick** [comment]
Just did a little research yesterday:

Jodel changed something inside the libhmac.so so radare2 isn't able to resolve the function name anymore. IDA Pro still is, but i'm too lazy to change the script to IDA. Retrieving the HMAC-Key is still possible using my [IDA-Keyhack](https://github.com/Unbrick/Jodel-Keyhack-v2) or just by [hooking into the java.security.Key class using frida](https://gist.github.com/Unbrick/c7151e44c4abf37cc0a6bc9d850b6a4a).

Additionally jodel changed the input given to the HMAC-Signing routine. Example of input of user registration and upvoting right here: 

Upvoting:  
```
PUT
%
api.go-tellm.com
%
443
%
/api/v2/posts/5d239cac23e3e1001bfbb1cf/upvote
%
12259522-a84741ac-a4e89ecc-acdc-4aba-aaf5-4e51572ea674
%
49.8831;8.6690
%
2019-07-08T19:44:29Z
%
home
%
false
%
{}
```

User registration:
Notice: The iid in the body of the requests is the same as the pushtoken used to verify the device using GCM.
```
POST
%
api.go-tellm.com
%
443
%
/api/v2/users/
%
%
%
2019-07-08T20:00:49Z
%
%
{"location":{"country":"DE","city":"....","loc_coordinates":{"lng":....,"lat":....},"loc_accuracy":15.21},"iid":"....","client_id":"81e8a76e-1e02-4d17-9ba0-8a7020261b26","registration_data":{"channel":"","referrer_id":"","campaign":"","feature":"","provider":"branch.io","referrer_branch_id":""},"device_uid":"....","language":"de-DE"}
```

Additionally somehow the gcmhack doesent work properly anymore, somehow it is not able to parse the protobuff stuff received by gcm, might update my jodel_api later with a working version.

**Comment by Unbrick** [comment]
Update: Fixed this project by not searching for function names anymore but instead searching for instructions and function length. Takes slightly more time to extract the key but works.

Fixed with commit  @697be97bb738cfb26c97af78d27c026dcac1c54e
